agent/app/agent_capability_test.go

```diff
@@ -1558,3 +1558,36 @@ func TestAppendGMSADomainlessCapabilitiesFalse(t *testing.T) {
 
 	assert.Equal(t, len(expectedCapabilities), len(capabilities))
 }
+
+func TestAppendFaultInjectionCapabilities(t *testing.T) {
+	originalIsnetworkToolingAvailable := isNetworkToolingAvailable
+	defer func() { isNetworkToolingAvailable = originalIsnetworkToolingAvailable }()
+
+	// Test case where required tooling is available
+	isNetworkToolingAvailable = func() bool { return true }
+	capabilities := []*ecs.Attribute{}
+	agent := &ecsAgent{}
+	capabilities = agent.appendFaultInjectionCapabilities(capabilities)
+	found := false
+	for _, attr := range capabilities {
+		if aws.StringValue(attr.Name) == "ecs.capability.fault-injection" {
+			found = true
+			break
+		}
+	}
```

Comment:
nit: since `appendFaultInjectionCapabilities` only appends a single capability, do we need this for-loop? Maybe just check that length is 1 and verify the only element?

Reply:
Implemented in https://github.com/aws/amazon-ecs-agent/commit/1a3ea5b7f2b55d50e8b41c84c10a75842c262893